webtranslate/users/development.py

```diff
@@ -3,8 +3,92 @@
 
 In particular the user management is very poor.
 """
+import os, configparser
 from webtranslate import rights, userauth
-from webtranslate.users import silly
+
+# Table with user names and plain text passwords.
+_users = None
+
+FILENAME = "users.dat"
+
+def init_users():
+    """
+    Initialize the user authentication system.
+    """
+    global _users
+
+    _users = set()
+    if not os.path.isfile(FILENAME): return
+
+    handle = open(FILENAME, "r", encoding = "utf-8")
+    for line in handle:
+        line = line.rstrip()
+        if len(line) == 0 or line[0] == '#': continue
+        i = line.find(':')
+        if i < 0: continue
+        j = line.find(':', i + 1)
+        if j < 0:
+            j = len(line)
+        uname, pwd = line[:i].strip(), line[i+1:j]
+
+        if len(uname) == 0 or len(pwd) == 0: continue
+        _users.add((uname, pwd))
+
+    handle.close()
+
+def authenticate(user, pwd):
+    """
+    Is the provided user and password combination valid?
+
+    @param user: User name.
+    @type  user: C{str}
+
+    @param pwd: Password.
+    @type  pwd: C{str}
+
+    @return: Whether the combination is valid (C{True} means 'valid').
+    @rtype:  C{bool}
+    """
+    global _users
+
+    if len(pwd) == 0: return False
+    return (user, pwd) in _users
+
+
+# Table with project user data.
+# Mapping of project names to mapping of roles to users.
+# C{dict} of C{str} to C{dict} of C{str} to C{set} of C{str}
+_projects = {}
+
+PROJECTSFILE = "projects.dat"
+
+def init_projects():
+    """
+    Initialize the projects table (mapping of projects to mapping of roles to users).
+    """
+    global _projects
+
+    # Read projects user data.
+    cfg = configparser.ConfigParser()
+    cfg.optionxform = lambda option: option # Don't convert keys to lowercase.
+    cfg.read(PROJECTSFILE)
+    _projects = {}
+    for pn in cfg.sections():
+        ps = cfg[pn]
+        values = {}
+        for k, ns in ps.items():
+            names = set()
+            for ns2 in ns.split(','):
+                for ns3 in ns2.split(' '):
+                    ns3 = ns3.strip()
+                    if len(ns3) < 3:
+                        # User names should be longer-equal to 3 characters.
```

Comment:
Lot of this code could use some more comments, but this is one that is not functional; the code literially says that. I miss the "why" this should be the case here :)